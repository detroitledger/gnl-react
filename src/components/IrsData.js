import React, { PropTypes } from 'react';

// Presentational component to display IRS data from SearchResult
const IrsData = ( props ) => {
  return (
    <div>
      <p>EIN: {props.irsOrganization.ein}</p>
      <p>Filing type: {props.irsOrganization.filing_type}</p>
      <p>Tax period: {props.irsOrganization.tax_period}</p>
      <p>Ledger name(s): {props.orgName}</p>
    </div>
  );
};

IrsData.propTypes = {
  irsOrganization: PropTypes.object.isRequired
};

export default IrsData;