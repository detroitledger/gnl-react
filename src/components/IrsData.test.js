import IrsData from './IrsData.js';
import React from 'react';
import renderer from 'react-test-renderer';

// this test passes, but expected jest to export a .snap file in this directory??
test('IrsData component renders irsOrganization data', () => {
  it('renders correctly', () => {
    const irsOrganization = { id: 1, ein: '111222333', filing_type: '990', total_revenue: 300 };
    const rendered = renderer.create(
  	  <IrsData irsOrganization={irsOrganization} />
  	);
  	expect(rendered.toJSON()).toMatchSnapshot();
  });
});