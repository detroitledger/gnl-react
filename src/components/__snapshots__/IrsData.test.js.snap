// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IrsData component renders irsOrganization data 1`] = `
<div>
  <p>
    This organization files a Form 
    990
     with the IRS.
  </p>
  <table>
    <thead>
      <tr>
        <th>
          Totals for tax period ending
           
          Jun 2012
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Revenue
        </td>
        <td>
          $300
        </td>
      </tr>
      <tr>
        <td>
          Expenses
        </td>
        <td>
          $301
        </td>
      </tr>
      <tr>
        <td>
          Assets
        </td>
        <td>
          $302
        </td>
      </tr>
      <tr>
        <td>
          Liabilities
        </td>
        <td>
          $303
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
